TYPO3.Neos/Classes/TYPO3/Neos/EventLog/Domain/Repository/EventRepository.php

```diff
@@ -59,9 +59,7 @@ public function removeAll()
         $classMetaData = $this->entityManager->getClassMetadata($this->getEntityClassName());
         $connection = $this->entityManager->getConnection();
         $databasePlatform = $connection->getDatabasePlatform();
-        $connection->query('SET FOREIGN_KEY_CHECKS=0');
         $truncateTableQuery = $databasePlatform->getTruncateTableSql($classMetaData->getTableName());
         $connection->executeUpdate($truncateTableQuery);
-        $connection->query('SET FOREIGN_KEY_CHECKS=1');
```

Comment:
Yes, also works. The documentation (http://dev.mysql.com/doc/refman/5.5/en/truncate-table.html) says:

> Foreign key constraints between columns of the same table are permitted. 

For PostgreSQL it's more or less the same (https://mariadb.com/kb/en/mariadb/truncate-table/):

> Constraints between columns in the same table are permitted.